pinot-core/src/main/java/org/apache/pinot/core/query/selection/SelectionOperatorUtils.java

```diff
@@ -94,27 +94,26 @@ public static List<ExpressionContext> extractExpressions(QueryContext queryConte
     }
 
     List<ExpressionContext> selectExpressions = queryContext.getSelectExpressions();
-    if (selectExpressions.size() == 1 && selectExpressions.get(0).equals(IDENTIFIER_STAR)) {
-      // For 'SELECT *', sort all columns (ignore columns that start with '$') so that the order is deterministic
-      Set<String> allColumns = indexSegment.getColumnNames();
-      List<String> selectColumns = new ArrayList<>(allColumns.size());
-      for (String column : allColumns) {
-        if (column.charAt(0) != '$') {
-          selectColumns.add(column);
-        }
-      }
-      selectColumns.sort(null);
-      for (String column : selectColumns) {
-        ExpressionContext expression = ExpressionContext.forIdentifier(column);
-        if (!expressionSet.contains(expression)) {
-          expressions.add(expression);
+    for (ExpressionContext selectExpression : selectExpressions) {
```

Comment:
This part seems unrelated to this PR. This rewrite should already be performed on the broker side, so shouldn't be needed